src/metaprob/examples/gaussian-notebook.ipynb

```diff
@@ -0,0 +1,260 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "(define example-2D\n",
+    "  (gen []\n",
+    "    (define x (gaussian 0 1))\n",
+    "    (define y (gaussian x 1))\n",
+    "    (+ x y)))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "(define get-trace\n",
+    "    (gen [proc inputs]\n",
+    "         (define out (empty-trace))\n",
+    "         (infer :procedure proc\n",
+    "                :inputs inputs\n",
+    "                :output-trace out)\n",
+    "         out))\n",
+    "\n",
+    "(define get-example-2D-traces\n",
+    "    (gen [n]\n",
+    "         (take n (repeatedly (gen [] (get-trace example-2D nil))))))\n",
+    "\n",
+    "(define get-xs\n",
+    "    (gen [traces]\n",
+    "         (map (gen [rho] (trace-get rho (addr 0 \"x\" \"gaussian\"))) traces)))\n",
+    "\n",
+    "(define get-ys\n",
+    "    (gen [traces]\n",
+    "         (map (gen [rho] (trace-get rho (addr 1 \"y\" \"gaussian\"))) traces)))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "(define make-example-2D-trace \n",
+    "    (gen [x y]\n",
+    "        (trace 0 (** (trace \"x\" (** (trace \"gaussian\" x))))\n",
+    "               1 (** (trace \"y\" (** (trace \"gaussian\" y))))\n",
+    "               )))\n",
+    "\n",
+    "(define score-example-2D\n",
+    "    (gen [x y]\n",
+    "         (get (infer :procedure example-2D\n",
+    "                     :inputs nil\n",
+    "                     :target-trace (make-example-2D-trace x y))\n",
+    "              2)))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "(.createBufferedImage\n",
+    "    (incanter.charts/heat-map score-example-2D\n",
+    "                              -3 3 -3 3\n",
+    "                              :title \"Bivariate Gaussian --- exact density\")\n",
+    "    300 300)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "(define example-2D-traces (get-example-2D-traces 1000))\n",
+    "(.createBufferedImage\n",
+    "    (incanter.charts/scatter-plot\n",
+    "        (get-xs example-2D-traces)\n",
+    "        (get-ys example-2D-traces)\n",
+    "        :title \"Bivariate Gaussian generative procedure --- sampled traces\")\n",
+    "    500 500)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "(define make-example-2D-density-y-given-x\n",
+    "    (gen [xval]\n",
+    "         (gen [y]\n",
+    "             (get (infer :procedure example-2D\n",
+    "                         :inputs nil\n",
+    "                         :target-trace (make-example-2D-trace xval y))\n",
+    "                  2))\n",
+    "         ))\n",
```

Comment:
Trailing parentheses should be on a single line instead of distinct lines. ([link](https://github.com/bbatsov/clojure-style-guide#gather-trailing-parens))

Reply:
I don't recognize "shoulds" without some kind of justification (for which I do not automatically _exclude_ appeals to authority). This advice seems to lack both rationale and authority. Does Mr. Batsov have exceptional standing in some way for making this kind of pronouncement?

Reply:
@jar398 Vikash specifically asked for feedback on style. The referenced style guide is the one that most of the Clojure community conforms to. Please assume that all feedback on style-related issues comes with an implicit "…if we want to conform to Clojure norms as expressed in bbatsov's style guide." ☺️